### PR TITLE
Add error handling conventions

### PR DESCRIPTION
**Pull request by janhalama**
## Description

This PR adds error handling conventions.

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have read the **CONTRIBUTING** document.
- [ ] I haven't repeated the code. (DRY)
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: CONVENTIONS.md [hunk 1]
```diff
@@ -49,6 +49,107 @@ We want first gather more content to understand how Comlink maps are written to
...
+Err {
+  "error": "HTTPError: HTTP Error
+AST Path: definitions[1].statements[0]
+Original Map Location: Line 44, column 3",
+}
+```
+
+The error output lacks information about status code and body of the response.
+
+You should handle unknown responses in map and map them to error model with HTTP status code and body of the request.
```
Comment (tmladek) [inline CONVENTIONS.md:110]:
```suggestion
You should handle unknown responses in the map, and map them to the error model, including the HTTP status code and the body of the response in the use case return value.
```

Not sure if that's your intended meaning (?)

File: CONVENTIONS.md [hunk 2]
```diff
@@ -49,6 +49,107 @@ We want first gather more content to understand how Comlink maps are written to
...
+  "title": "Unknown error",
+  "detail": "Status code: 404, body: \\"Not found\\"",
+  "code": "UnknownError"
+}
+AST Path: definitions[1].statements[0].responseHandlers[1].statements[1]
+Original Map Location: Line 64, column 7",
+}
+```
+
+Be aware that handling HTTP 500 status codes will prevent OneSDK to failover to backup provider. HTTP 500 status codes handling is not recommended for commodity use cases like sending e-mails, sms messages, getting weather forecast,...
```
Comment (tmladek) [inline CONVENTIONS.md:151]:
```suggestion
Be aware that **handling HTTP 500 status codes will prevent OneSDK failover** to a backup provider. Handling HTTP 500 status codes is not recommended for commodity use cases like sending e-mails, SMS messages, getting weather forecast, etc.
```

**Comment by janhalama** [comment]
@tmladek thank you for your suggestions 🙏

**Comment by janhalama** [comment]
The error handling conventions are obsolete with the new WASM OneSDK.

**Pull request closed without merging**